samples/csharp_dotnetcore/51.cafe-bot/Dialogs/Shared/Prompts/GetLocationDateTimePartySizePrompt.cs

```diff
@@ -0,0 +1,175 @@
+        private const string ConfirmCancelPrompt = "confirmCancelPrompt";
+
+        // LUIS service type entry for turn.n book table LUIS model in the .bot file.
+        private static readonly string LuisConfiguration = MsBotName + "_" + "cafeBotBookTableTurnNModel";
+
+        private readonly BotServices _botServices;
+        private readonly IStatePropertyAccessor<UserProfile> _userProfileAccessor;
+        private readonly IStatePropertyAccessor<OnTurnProperty> _onTurnAccessor;
+        private readonly IStatePropertyAccessor<ReservationProperty> _reservationsAccessor;
+
+        public GetLocationDateTimePartySizePrompt(
+                    string dialogId,
+                    BotServices botServices,
+                    IStatePropertyAccessor<ReservationProperty> reservationsAccessor,
+                    IStatePropertyAccessor<OnTurnProperty> onTurnAccessor,
+                    IStatePropertyAccessor<UserProfile> userProfileAccessor,
+                    PromptValidator<string> validator = null)
+            : base(dialogId, validator)
+        {
+            _botServices = botServices ?? throw new ArgumentNullException(nameof(botServices));
+            _userProfileAccessor = userProfileAccessor ?? throw new ArgumentNullException(nameof(botServices));
+            _onTurnAccessor = onTurnAccessor ?? throw new ArgumentNullException(nameof(onTurnAccessor));
+            _reservationsAccessor = reservationsAccessor ?? throw new ArgumentNullException(nameof(reservationsAccessor));
+        }
+
+        public async override Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var turnContext = dc.Context;
+            var step = dc.ActiveDialog.State;
+
+            // get reservation property
+            var newReservation = await _reservationsAccessor.GetAsync(turnContext, () => new ReservationProperty());
+
+            // Get on turn property. This has any entities that mainDispatcher,
+            //  or Bot might have captured in its LUIS model
+            var onTurnProperties = await _onTurnAccessor.GetAsync(turnContext, () => new OnTurnProperty("None", new List<EntityProperty>()));
+
+            // if on turn property has entities
+            ReservationResult updateResult = null;
+            if (onTurnProperties.Entities.Count > 0)
+            {
+                // update reservation property with on turn property results
+                updateResult = newReservation.UpdateProperties(onTurnProperties);
+            }
+
+            // see if updates to reservation resulted in errors, if so, report them to user.
+            if (updateResult != null &&
+                updateResult.Status == ReservationStatus.Incomplete &&
+                updateResult.Outcome != null &&
+                updateResult.Outcome.Count > 0)
+            {
+                // set reservation property
```

Comment:
>set reservation property [](start = 19, length = 24)

set =>  Set, missing period.  Most comments in this file need this change, please review.